src/test/exporter/test_exporter.cc

```diff
@@ -692,3 +726,706 @@ TEST(Exporter, check_labels_and_metric_name) {
+# TYPE ceph_rgw_op_put_obj_lat_sum gauge
+ceph_rgw_op_put_obj_lat_sum{Bucket="bucket1",instance_id="aayrrj"} 2.818065
+ceph_rgw_op_put_obj_lat_sum{User="dashboard",instance_id="aayrrj"} 0.000000
+# HELP ceph_rgw_op_put_obj_ops Puts
+# TYPE ceph_rgw_op_put_obj_ops counter
+ceph_rgw_op_put_obj_ops{Bucket="bucket1",instance_id="aayrrj"} 2
+ceph_rgw_op_put_obj_ops{User="dashboard",instance_id="aayrrj"} 0
+)";
+
+  ASSERT_TRUE(collector.metrics.find(expectedMetrics) != std::string::npos);
+
+  // Test for labeled metrics - RBD_MIRROR
+  daemon = "rbd-mirror.a";
+  setup_test_data(clients, daemon, LABELED_RBD_MIRROR, collector);
+
+  expectedMetrics = R"(
+# HELP ceph_rbd_mirror_snapshot_image_last_sync_bytes Bytes synced for the last snapshot
+# TYPE ceph_rbd_mirror_snapshot_image_last_sync_bytes gauge
+ceph_rbd_mirror_snapshot_image_last_sync_bytes{ceph_daemon="rbd-mirror.a",image="image1",namespace="",pool="data"} 52428800
+# HELP ceph_rbd_mirror_snapshot_image_last_sync_time Time taken to sync the last snapshot
+# TYPE ceph_rbd_mirror_snapshot_image_last_sync_time gauge
+ceph_rbd_mirror_snapshot_image_last_sync_time{ceph_daemon="rbd-mirror.a",image="image1",namespace="",pool="data"} 0.194676
+# HELP ceph_rbd_mirror_snapshot_image_local_timestamp Timestamp of the local snapshot
+# TYPE ceph_rbd_mirror_snapshot_image_local_timestamp gauge
+ceph_rbd_mirror_snapshot_image_local_timestamp{ceph_daemon="rbd-mirror.a",image="image1",namespace="",pool="data"} 1702884232.559627
+# HELP ceph_rbd_mirror_snapshot_image_remote_timestamp Timestamp of the remote snapshot
+# TYPE ceph_rbd_mirror_snapshot_image_remote_timestamp gauge
+ceph_rbd_mirror_snapshot_image_remote_timestamp{ceph_daemon="rbd-mirror.a",image="image1",namespace="",pool="data"} 1702884232.559627
+# HELP ceph_rbd_mirror_snapshot_image_snapshots Number of snapshots synced
+# TYPE ceph_rbd_mirror_snapshot_image_snapshots counter
+ceph_rbd_mirror_snapshot_image_snapshots{ceph_daemon="rbd-mirror.a",image="image1",namespace="",pool="data"} 1
+# HELP ceph_rbd_mirror_snapshot_image_sync_bytes Total bytes synced
+# TYPE ceph_rbd_mirror_snapshot_image_sync_bytes counter
+ceph_rbd_mirror_snapshot_image_sync_bytes{ceph_daemon="rbd-mirror.a",image="image1",namespace="",pool="data"} 52428800
+# HELP ceph_rbd_mirror_snapshot_image_sync_time_count Average sync time Count
+# TYPE ceph_rbd_mirror_snapshot_image_sync_time_count counter
+ceph_rbd_mirror_snapshot_image_sync_time_count{ceph_daemon="rbd-mirror.a",image="image1",namespace="",pool="data"} 1
+# HELP ceph_rbd_mirror_snapshot_image_sync_time_sum Average sync time Total
+# TYPE ceph_rbd_mirror_snapshot_image_sync_time_sum gauge
+ceph_rbd_mirror_snapshot_image_sync_time_sum{ceph_daemon="rbd-mirror.a",image="image1",namespace="",pool="data"} 0.194676
+)";
+  ASSERT_TRUE(collector.metrics.find(expectedMetrics) != std::string::npos);
+}
+
+TEST(Exporter, add_fixed_name_metrics) {
+  std::string metric_name = "ceph_data_sync_from_zone2-zg1-realm1_fetch_bytes";
```

Comment:
Can we have more test cases?

Reply:
Will add a couple more

Reply:
I've added a few more for metrics name used in RGW grafana dashboards